src/badvpn/tun2socks/tun2socks.c

```diff
@@ -1251,9 +1416,9 @@ int process_device_udp_packet (uint8_t *data, int data_len)
 
             // if transparent DNS is enabled, any packet arriving at out netif
             // address to port 53 is considered a DNS packet
-            is_dns = (options.udpgw_transparent_dns &&
-                      ipv4_header.destination_address == netif_ipaddr.ipv4 &&
-                      udp_header.dest_port == hton16(53));
+            is_dns = (options.transparent_dns &&
+                      // ipv4_header.destination_address == netif_ipaddr.ipv4 &&
```

Comment:
I had commented out this line to check if we were receiving non-DNS UDP packets.
It should be uncommented.